### ISSUE TITLE
DroidFS PDF viewer does not display file, crashes app

### ISSUE THREAD

**Issue by r9591** [hardcore-sushi/DroidFS#303]
DroidFS 2.2.0 installs without issue but PDF viewer does not load file then crashes the app. 

I have a log file, how do I send it to you?

I'm new on github, have used DroidFS a long time,
had to sign up here to let you know. The app is
amazing but the PDF thing was disappointing.

Thank you for the app.

**Comment by hardcore-sushi** [comment]
It would be very helpful if you can share the specific PDF file causing the crash. You can send it with your crash log here as GitHub attachments. You can encrypt them with my PGP key if they contain sensitive info.

> The install was interupted because it would not upgrade on top of a previous version, so I need to uninstall the old version first.

Where did you download the previous and the new versions from?

> had to sign up here to let you know

The repository is also hosted on [Gitea](https://forge.chapril.org/hardcoresushi/DroidFS) specifically for people wanting to avoid GitHub.

**Comment by r9591** [comment]
I cant remember where the previous version came from, but very probably F-droid. The recent upgrade was from here at Gihub.

I'm sending the log and pdf. The pdf is a random junk document i downloaded, from where i don't recall, to test the app.

Thanks for the speedy reply.

[DroidFS_2024-08-25_04_10_51.log](https://github.com/user-attachments/files/16742286/DroidFS_2024-08-25_04_10_51.log)

[NPPF_December_2023.pdf](https://github.com/user-attachments/files/16742288/NPPF_December_2023.pdf)

**Comment by hardcore-sushi** [comment]
> I cant remember where the previous version came from, but very probably F-droid. The recent upgrade was from here at Gihub.

Then that's expected. F-Droid builds are signed with F-Droid keys, whereas GitHub and Gitea builds are signed by me.

The app doesn't crash on my side. According to the provided logs, the app seems to have crashed when you clicked on the "Properties" menu in the PDF viewer screen. Is this correct? Or did the app crash as soon as you opened the file without any further interaction?

**Comment by r9591** [comment]
Yes, Properties, because the file did not display, white screen only, so Properties then crash!

Should I go back to F-droid and try their version?

**Comment by hardcore-sushi** [comment]
OK. Could you share your logcat recorded while you open the PDF file?

You can try the F-Droid version but I think it will be the same result. What's your Android ROM? What's the version of your WebView?

**Comment by r9591** [comment]
Where do I get a logcat?

I'm using a moto e13, Model number (SKU): XT2345-3,
Hardware version: PVT

Kernel:
5.4.190-android12-9-g80224059bda2-ab313

**Comment by hardcore-sushi** [comment]
DroidFS has a built-in logcat viewer in the settings. Just open the PDF file then go to this screen, click save in the top right corner and share the saved file. Otherwise you can use [LogFox](https://f-droid.org/en/packages/com.f0x1d.logfox) if you have root access.

Can you go to Developer options > WebView implementation and show your system webview version?

**Comment by r9591** [comment]
I found the Webview version: 

108.0.5359.128

I found the logcat, will send that to you.

**Comment by r9591** [comment]
Here is the logcat

[DroidFS_2024-08-26_20_13_15.log](https://github.com/user-attachments/files/16752910/DroidFS_2024-08-26_20_13_15.log)

**Comment by hardcore-sushi** [comment]
Thanks. That version is pretty old. It was released almost 2 years ago. Using a newer webview may fix the issue.

Your second logcat is unfortunately not helpful as it does not seem to include any related warnings.

**Comment by r9591** [comment]
The analysis sounds good, i agree, very likely old Webview.

I have now uninstalled version 2.2.0 and replaced it with 2.1.3 from your Github until i get Webview fixed. 

Everything appears to be working now so all is well.

Thank you

**Comment by hardcore-sushi** [comment]
Oh so it works with 2.1.3. As the DroidFS PDF viewer is based on the [GrapheneOS one](https://github.com/GrapheneOS/PdfViewer), can you try to open your file with the original PdfViewer app version 17, 18 and 19? You can download the APKs from their release page: https://github.com/GrapheneOS/PdfViewer/releases
